<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1122</width>
    <height>654</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1121</width>
      <height>631</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(240, 240, 240);</string>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="insert_delete_page">
     <attribute name="title">
      <string>插入删除</string>
     </attribute>
     <widget class="QFrame" name="course_frame">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="course_insert_button">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>220</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>插入</string>
       </property>
      </widget>
      <widget class="QPushButton" name="course_delete_button">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>220</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;删除时仅课程号有效&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="whatsThis">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>删除</string>
       </property>
      </widget>
      <widget class="QLabel" name="course_label">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>10</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font-weight:bold;</string>
       </property>
       <property name="text">
        <string>课程信息</string>
       </property>
      </widget>
      <widget class="QWidget" name="formLayoutWidget_2">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>331</width>
         <height>61</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="course_number_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>课程号：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="course_name_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>课程名称：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="course_number_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="course_name_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_4">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>351</width>
         <height>261</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">border:1px solid #8f8f8f;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <zorder>frame_4</zorder>
      <zorder>course_insert_button</zorder>
      <zorder>course_delete_button</zorder>
      <zorder>course_label</zorder>
      <zorder>formLayoutWidget_2</zorder>
     </widget>
     <widget class="QFrame" name="student_frame">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="student_insert_button">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>220</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>插入</string>
       </property>
      </widget>
      <widget class="QPushButton" name="student_delete_button">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>220</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;删除时仅学号有效&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>删除</string>
       </property>
      </widget>
      <widget class="QLabel" name="student_label">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>10</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font-weight:bold;</string>
       </property>
       <property name="text">
        <string>学生信息</string>
       </property>
      </widget>
      <widget class="QWidget" name="formLayoutWidget">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>331</width>
         <height>151</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="student_number_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>学号：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="sutdent_age_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>年龄：</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="student_sex_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>性别：</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_8">
          <property name="styleSheet">
           <string notr="true">;</string>
          </property>
          <property name="text">
           <string>年级：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="student_name_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>姓名：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="student_number_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="student_name_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="student_age_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLineEdit" name="student_sex_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLineEdit" name="student_grade_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_5">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>351</width>
         <height>261</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">border:1px solid #8f8f8f;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <zorder>frame_5</zorder>
      <zorder>student_insert_button</zorder>
      <zorder>student_delete_button</zorder>
      <zorder>student_label</zorder>
      <zorder>formLayoutWidget</zorder>
     </widget>
     <widget class="QFrame" name="take_frame">
      <property name="geometry">
       <rect>
        <x>750</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="take_insert_button">
       <property name="geometry">
        <rect>
         <x>60</x>
         <y>220</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>插入</string>
       </property>
      </widget>
      <widget class="QPushButton" name="take_delete_button">
       <property name="geometry">
        <rect>
         <x>200</x>
         <y>220</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;删除时仅学号和课程号有效&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
       </property>
       <property name="text">
        <string>删除</string>
       </property>
      </widget>
      <widget class="QLabel" name="take_label">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>10</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font-weight:bold;</string>
       </property>
       <property name="text">
        <string>选课信息</string>
       </property>
      </widget>
      <widget class="QWidget" name="formLayoutWidget_3">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>331</width>
         <height>91</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="take_student_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>学号：</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="take_course_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>课程号：</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="take_grade_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>成绩：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="take_student_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="take_course_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="take_grade_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_7">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>351</width>
         <height>261</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">border:1px solid #8f8f8f;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <zorder>frame_7</zorder>
      <zorder>take_insert_button</zorder>
      <zorder>take_delete_button</zorder>
      <zorder>take_label</zorder>
      <zorder>formLayoutWidget_3</zorder>
     </widget>
     <widget class="QTableView" name="insert_delete_table">
      <property name="geometry">
       <rect>
        <x>5</x>
        <y>281</y>
        <width>1101</width>
        <height>311</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="query_name">
     <attribute name="title">
      <string>查询</string>
     </attribute>
     <widget class="QFrame" name="join_query">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="join_query_button">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>230</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>查询</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>10</y>
         <width>171</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font-weight:bold;</string>
       </property>
       <property name="text">
        <string>查询学生已选课程</string>
       </property>
      </widget>
      <widget class="QWidget" name="formLayoutWidget_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>321</width>
         <height>31</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_4">
        <item row="0" column="0">
         <widget class="QLabel" name="join_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>学号：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="join_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>351</width>
         <height>261</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">border:1px solid #8f8f8f;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <zorder>frame</zorder>
      <zorder>join_query_button</zorder>
      <zorder>label</zorder>
      <zorder>formLayoutWidget_4</zorder>
     </widget>
     <widget class="QFrame" name="nested_query">
      <property name="geometry">
       <rect>
        <x>380</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="nested_query_button">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>120</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>查询</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>10</y>
         <width>151</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font-weight:bold;</string>
       </property>
       <property name="text">
        <string>查询不住校学生</string>
       </property>
      </widget>
      <widget class="QFrame" name="frame_2">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>351</width>
         <height>261</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">border:1px solid #8f8f8f;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <zorder>frame_2</zorder>
      <zorder>nested_query_button</zorder>
      <zorder>label_2</zorder>
     </widget>
     <widget class="QFrame" name="group_query">
      <property name="geometry">
       <rect>
        <x>750</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <widget class="QPushButton" name="group_query_button">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>230</y>
         <width>93</width>
         <height>28</height>
        </rect>
       </property>
       <property name="text">
        <string>查询</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>80</x>
         <y>10</y>
         <width>191</width>
         <height>31</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑</family>
         <pointsize>12</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">font-weight:bold;</string>
       </property>
       <property name="text">
        <string>查询一定人数的社团</string>
       </property>
      </widget>
      <widget class="QWidget" name="formLayoutWidget_6">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>321</width>
         <height>31</height>
        </rect>
       </property>
       <layout class="QFormLayout" name="formLayout_6">
        <item row="0" column="0">
         <widget class="QLabel" name="group_label">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>人数下限：</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="group_text">
          <property name="styleSheet">
           <string notr="true">background-color: rgb(255, 255, 255);</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QFrame" name="frame_8">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>351</width>
         <height>261</height>
        </rect>
       </property>
       <property name="styleSheet">
        <string notr="true">border:1px solid #8f8f8f;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
      </widget>
      <zorder>frame_8</zorder>
      <zorder>group_query_button</zorder>
      <zorder>label_3</zorder>
      <zorder>formLayoutWidget_6</zorder>
     </widget>
     <widget class="QTableView" name="query_table">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>290</y>
        <width>1101</width>
        <height>311</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">background-color: rgb(255, 255, 255);</string>
      </property>
     </widget>
     <widget class="QFrame" name="frame_3">
      <property name="geometry">
       <rect>
        <x>750</x>
        <y>10</y>
        <width>351</width>
        <height>261</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">border:1px solid #d6d6d6;</string>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
     <zorder>frame_3</zorder>
     <zorder>join_query</zorder>
     <zorder>nested_query</zorder>
     <zorder>group_query</zorder>
     <zorder>query_table</zorder>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1122</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>事务管理</string>
    </property>
    <addaction name="commit"/>
    <addaction name="rollback"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <action name="commit">
   <property name="text">
    <string>提交</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="rollback">
   <property name="text">
    <string>回滚</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
